src/ui/components/JEXLDebuggerPage.tsx

```diff
@@ -100,27 +205,26 @@ const JEXLDebuggerPage: FC = () => {
           >
             Refresh Context
           </Button>
-          {memoizedClientContextEntries.map(([key, value]) => (
+          {Object.entries(modifiedContext).map(([key]) => (
             <Row key={key} className="mb-4 d-flex align-items-center">
               <Col xs={3} className="secondary-fg fw-bold">
                 {key}
               </Col>
               <Col xs={9}>
-                {["number", "string", "boolean"].includes(typeof value) ? (
-                  <Form.Control
-                    type="text"
-                    readOnly
-                    value={String(value)}
-                    className="p-3 w-50 grey-border short-text"
-                  />
-                ) : (
-                  <Form.Control
-                    as="textarea"
-                    readOnly
-                    value={JSON.stringify(value, null, 2)}
-                    className="p-3 w-50 grey-border long-text"
-                  />
-                )}
+                <ContextField
+                  contextKey={key}
+                  contextValue={
+                    (formData[key] ?? modifiedContext[key]) as ContextValue
+                  }
+                  onChange={(key, value) =>
+                    handleContextChange(
+                      key,
+                      String(value),
+                      getFieldType(originalContext[key]),
+                    )
+                  }
```

Comment:
This creates a new fn every time this component renders, which causes the child component to re-render, even if there is no change.

We're passing the fieldType though and the child component knows the key and the value, so it should be responsible for calling useCallback() to create a memoized callback.